packages/airnode-adapter/src/types.ts

```diff
@@ -78,3 +78,48 @@ export interface ExtractedAndEncodedResponse {
 }
 
 export type ReservedParametersDelimeter = typeof MULTIPLE_PARAMETERS_DELIMETER | typeof PATH_DELIMETER;
+
+export class SizeLimitError extends Error {
```

Comment:
I dislike these custom error messages and we do not use this pattern in Airnode (and I think it's not that common in JS/TS world in general). I think we should just use the `Error` with a different error messages.

Reply:
I noticed that this pattern hasn't been used but how would you prefer to check for the type of error in the `call-api.ts` `extractAndEncodeResponse` catch-block in this case? Checking for the error (message) string content doesn't seem like a great alternative? Would you have a better suggestion?

Reply:
Dealing and parsing strings is something that I have regretted many times in the past. A alternative could be to return a new AirnodeError object that has an error code and the Error object inside. Then you could use the error code to determine which error message to display to the user. Just an idea.

Reply:
I like this solution as well @acenolaza. @Siegrift What do you think, should we go with that?

Reply:
I agree with @Siegrift I'm also not really sure about this approach... 

If we're going to be relying on error responses so much, maybe it could be better to wrap the responses in an object instead of throwing and then matching on different error types (?)

```ts
{ 
  error: { 
    type: 'SizeLimitError', // Some kind of enum
    message: 'optional custom message'
  },
  value: null
}
```

Just an idea though. What do you think?

Reply:
This also seems like a good alternative to me. I guess it depends on which solution everyone finds fits the best for this issue.

Reply:
Alternatively an even shorter solution (assuming we don't really care about specifying the errors in detail). We could do the following:

1. all errors thrown are instances of a normal `Error` with a string `message`
2. and then in _call-api.ts_:

```typescript
} catch (e) {
  if (e instanceof Error) {
      const log = logger.pend('ERROR', e.message);
      return [[log], { success: false, errorMessage: e.message }];
  }
  const log = logger.pend('ERROR', String(e));
  return [[log], { success: false, errorMessage: String(e) }];
}
```

Reply:
I'd do exactly that (just provider better error message). I'd also drop the instanceof check and always assume that we did throw an error (this holds in practice).

I am not a fan of the suggestion with custom error message nor wrapping, since they both require some centralized enum with allowed error codes, which doesn't work well with monorepo.